tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php

```diff
@@ -462,7 +473,28 @@ public function testEmptyUpdate(): void
         self::assertSame($qb2, $qb);
     }
 
-    public function testDelete(): void
+    public function testDeleteOnMysqlPlatforms(): void
+    {
+        $platform = $this->createMock(AbstractPlatform::class);
+        $platform
+            ->expects($this->any())
+            ->method('getRequestsAdditionalDeleteQueryTableAliasBeforeFrom')
+            ->willReturn(true);
+
+        $this->conn = $this->createMock(Connection::class);
+        $this->conn
+            ->expects($this->any())
+            ->method('getDatabasePlatform')
+            ->willReturn($platform);
+
+        $qb = new QueryBuilder($this->conn);
+        $qb->delete('users', 'u');
```

Comment:
The changes like this should be covered with an integration test first of all. Unit tests in this case are not that important IMO. Please see the [testing guildelines](https://www.doctrine-project.org/projects/doctrine-dbal/en/2.12/reference/testing.html).

Reply:
The changed and added unit test was more about making it work with the changes.
I'll add integration tests if we decide that we want this change.